integration-tests/hibernate-reactive-db2/src/main/java/io/quarkus/it/hibernate/reactive/db2/HibernateReactiveDB2AlternativeTestEndpoint.java

```diff
@@ -34,23 +34,25 @@
     public CompletionStage<GuineaPig> reactiveFind() {
         final GuineaPig expectedPig = new GuineaPig(5, "Aloi");
         return populateDB().convert().toCompletionStage()
-                .thenCompose(junk -> stageSession.find(GuineaPig.class, expectedPig.getId()));
+                .thenCompose(junk -> stageSessionFactory.withTransaction(
+                        (stageSession, transaction) -> stageSession.find(GuineaPig.class, expectedPig.getId())));
```

Comment:
That's nitpicking, but personally I'd rename both `stageSession` and `mutinySession` to just `session` everywhere. It's better if someone copy-pastes that code (and we know someone will...).